app/src/sharedTest/java/org/oppia/android/app/onboarding/CreateProfileFragmentTest.kt

```diff
@@ -0,0 +1,433 @@
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = CreateProfileFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class CreateProfileFragmentTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var editTextInputAction: EditTextInputAction
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  @Test
+  fun testFragment_nicknameLabelIsDisplayed() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            allOf(
+              isDisplayed(),
+              withText(
+                context.getString(
+                  R.string.create_profile_activity_nickname_label
+                )
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_nicknameEditTextIsDisplayed() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testFragment_stepCountText_isDisplayed() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.onboarding_steps_count))
+        .check(
+          matches(
+            allOf(
+              isDisplayed(),
+              withText(
+                context.getString(
+                  R.string.onboarding_step_count_three
+                )
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_continueButtonClicked_filledNickname_launchesLearnerIntroScreen() {
+    launchNewLearnerProfileActivity().use {
+      onView(withId(R.id.create_profile_nickname_edittext))
+        .perform(
+          editTextInputAction.appendText("John"),
+          closeSoftKeyboard()
+        )
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.onboarding_navigation_continue))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+
+      // No screen change as the navigation to the next screen is not implemented yet.
+      // This should fail in the future once the screen has been implemented.
+      onView(withId(R.id.create_profile_nickname_label))
+        .check(
+          matches(
+            withText(
+              context.getString(
+                R.string.create_profile_activity_nickname_label
+              )
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_continueButtonClicked_emptyNickname_showNicknameErrorText() {
```

Comment:
Probably would be nice also having a test to verify that the error text does NOT show if the nickname isn't empty.

Reply:
I added these assertions.